third_party/icu/googleurl/test.cc

```diff
@@ -0,0 +1,19 @@
+#include "url/gurl.h"
+
+#include "gtest/gtest.h"
+
+// Basic smoke test to ensure that GURL (with shimmed ICU) works properly.
+TEST(GoogleUrl, SmokeTest) {
+  GURL url("https://example.org/test?foo=bar#section");
+  EXPECT_TRUE(url.is_valid());
+  EXPECT_EQ(url.scheme(), "https");
+  EXPECT_EQ(url.host(), "example.org");
+  EXPECT_EQ(url.EffectiveIntPort(), 443);
+  EXPECT_EQ(url.path(), "/test");
+  EXPECT_EQ(url.query(), "foo=bar");
+  EXPECT_EQ(url.ref(), "section");
+
+  // Ensure ICU shim is functioning correctly, i.e. not crashing and resulting invalid parsed URL.
+  GURL idn_url("https://\xe5\x85\x89.example/");
```

Comment:
Can you also add tests that have %-encoded sequences in the host name as well as tests with hostnames that are not [DNS RFC](https://tools.ietf.org/html/rfc1034#section-3.5) compliant, please?

Reply:
I added some. but I think my creativity is lacking. If you have some examples, please let me know. Thank you!

Reply:
Btw, this seems like a great (and not so difficult) candidate for a fuzzer that would catch GURL crashes
@dio, what do you think? I think it would very roughly look like https://github.com/envoyproxy/envoy/blob/master/test/common/common/hash_fuzz_test.cc

Reply:
@asraa thanks! I added the setup. Please take a look when you have time.